imjoy/worker.py

```diff
@@ -0,0 +1,150 @@
+"""Provide worker functions."""
+import traceback
+import sys
+import time
+
+from .imjoyUtils import formatTraceback
+from .util import Registry
+
+try:
+    import queue
+except ImportError:
+    import Queue as queue
+
+# pylint: disable=unused-argument
+
+JOB_HANDLERS = Registry()
+
+
+def task_worker(conn, sync_q, logger, abort):
+    """Implement a task worker."""
+    while True:
+        if abort is not None and abort.is_set():
+            break
+        try:
+            job = sync_q.get()
+        except queue.Empty:
+            time.sleep(0.1)
+            continue
+        sync_q.task_done()
```

Comment:
Why do we say that the task is done, before we run the task? Shouldn't we run the task first and then say it's done?

Reply:
It doesn't matter here since we are not counting tasks in the queue or waiting for it to complete, but I agree that it should be better to call task_done after the job complete.
And I don't remember whether I did that on purpose or not, maybe we should not change this for the moment, and take a note to experimenting with it.

Reply:
Sounds good!